locations/pipelines/apply_nsi_categories.py

```diff
@@ -50,7 +50,13 @@ def process_item(self, item, spider):
         spider.crawler.stats.inc_value("atp/nsi/match_failed")
         return item
 
-    def filter_cc(self, matches, cc) -> []:
+    def filter_cc(self, matches: list[dict], cc: str, categories: dict = None) -> list[dict]:
+        """Filter matches by country code, attempt to find a better match if category is supplied.
+        :param matches: list of matches from NSI
+        :param cc: country code in lower case
+        :param categories: category tags
```

Comment:
Can the documentation for `categories` clarify that categories have to be an exact match (e.g. if two tags are supplied, both tags have to present on the NSI item for a match to occur)?

Reply:
(this strict matching is what the test cases are testing for in this proposed change)

Reply:
Suggest too that documentation also clarifies that the strict matching is only done for `top_level_tags` from categories.py and any other tags are ignored in the matching.

Reply:
Yep. This doc update makes sense in `filter_categories`.

Reply:
> Suggest too that documentation also clarifies that the strict matching is only done for top_level_tags from categories.py and any other tags are ignored in the matching.

Name `get_category_tags` is already tells us that only category tags are used.